ipm_library/ipm_library/ipm.py

```diff
@@ -76,8 +77,8 @@ def map_point(
             self,
             plane: Plane,
             point: Point2D,
-            time: Time,
-            plane_frame_id: str,
+            time: Time = Time(),
+            plane_frame_id: Optional[str] = None,
```

Comment:
> Additionally, instantiation inside the function signature is strongly discouraged, as the same object is created once and used for all future calls (see https://betterprogramming.pub/the-unexpected-workings-of-pythons-default-arguments-fe1a4addb1dc). This is fine for immutable default types (str, int, ...) and could be also fine for the Time class, but it should be avoided.

Didn't know this! Thanks for pointing me to that. I'll keep that in mind.

What about having ``time: Optional[Time] = None`` instead? I understand your point of the users not filling out the time parameter, and leaving it until later realising that they're getting some errors.

However, I'm in favour of the convenience of not having to enter a time but using the latest available transform by default. This makes tests a lot easier to read, and would allow the service to be tested through the ros2cli. As far as i know, you can't test the ipm service with ``ros2 service call`` because you can't obtain the time value easily in the cli.

I think this is where the tutorials we would write on https://github.com/ros-sports/ipm-docs could help. The main examples can fill out the time parameter, and explain to get an accurate transform, you need to put the time in.

Reply:
@Flova I think this is the last point to discuss now, whether to have a default value for ``Time``. @jaagut would you have some thoughts on this?

Reply:
I am not quite sure, what you want to achieve with `time: Optional[Time] = None`? I guess somewhere down the line you would need to handle the `None` case, as it is not a valid `Time` msg, that can be passed into the header's stamp: https://github.com/Flova/ipm/blob/9d9d717080e9bbf8773b953bf32f6766e6f4ffb7/ipm_library/ipm_library/utils.py#L72

I don't know how to handle `None` and thereby mitigating the problem @Flova stated in the beginning.

Reply:
Yes, code should be added to handle the ``None``, and ensure ``tf2::lookupTransform`` is called with Time(), to get the latest transform.

Reply:
I see your point, but except for the described static case, the time stamp of the measurement is always needed (I consider the default use of approximate times a bad practice in this case), so the code is not really longer. If you know what you are doing, you could still pass a `Time(0)` instance manually and get the latest transforms. We could also describe this in a tutorial. In addition to this, I think that in most cases the timestamp of the detection should be preserved for other processing down the line. If the latest transform is requested in the None (default) case, it is set to the stamp of the latest transform, which might be confusing down the line.

So overall I think this is a minor thing of the API and I would like to move forward. I would be more happy to do it without a default and a bit of docs for more advanced uses (e.g. latest common transform), but I don't have a too strong opinion in this regard.

Reply:
Ok, I'm going to remove my changes here and track this in https://github.com/ros-sports/ipm/issues/46